File: src/lib/openwig/ZCartridge.ts
import { ZObject } from './ZObject';

export class ZCartridge extends ZObject {
  public allZObjects: ZObject[] = [];
  public zones: any[] = [];
  public tasks: any[] = [];
  public inventory: any[] = [];

  constructor() {
    super();
    this.table['AllZObjects'] = this.allZObjects;
    this.table['Zones'] = this.zones;
    this.table['Tasks'] = this.tasks;
    this.table['Inventory'] = this.inventory;
  }

  public addObject(obj: ZObject) {
    this.allZObjects.push(obj);
  }
}

File: java-backport/net/jakedot/jourwigo/openwig/ZCartridge.java
package net.jakedot.jourwigo.openwig;

import java.util.ArrayList;
import java.util.List;

public class ZCartridge extends ZObject {
    public List<ZObject> allZObjects = new ArrayList<>();
    public List<ZZone> zones = new ArrayList<>();
    public List<Object> tasks = new ArrayList<>();
    public List<Object> inventory = new ArrayList<>();

    public ZCartridge() {
        super();
        table.put("AllZObjects", allZObjects);
        table.put("Zones", zones);
        table.put("Tasks", tasks);
        table.put("Inventory", inventory);
    }

    public void addObject(ZObject obj) {
        allZObjects.add(obj);
    }
}
